File: LeetCode/easy/Maximum_Subarray.cxx
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int res = nums.front();
        int cur = 0;
        for (auto x : nums)
        {
            if (cur + x > x)
            {
                cur += x;
            }
            else
            {
                cur = x;
            }
            res = std::max(res, cur);
        }
        return res;
    }
};

File: LeetCode/medium/Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = -111111111111
        l = 0
        r = 0

        while l < len(nums):
            r = l
            s = 0
            while r < len(nums) and s >= 0:
                s += nums[r]
                res = max(res, s)
                r += 1
            l = r
        return res


